glean-core/src/upload/directory.rs

```diff
@@ -45,87 +38,154 @@ fn get_file_name_as_str(path: &Path) -> Option<&str> {
-/// This vector will be ordered by file `modified_date`.
-///
-/// Any files that don't match the UUID regex will be deleted
-/// to prevent files from polluting the pings directory.
-///
-/// Files that are not correctly formatted will also be deleted.
-pub fn process_pings_dir(data_path: &Path) -> Result<Vec<PingRequest>> {
-    let pings_dir = get_pings_dir(data_path);
-    let mut pending_pings: Vec<_> = pings_dir
-        .read_dir()?
-        .filter_map(|entry| entry.ok())
-        .filter_map(|entry| {
-            let path = entry.path();
-            if let Some(file_name) = get_file_name_as_str(&path) {
-                // Delete file if it doesn't match the pattern.
-                if Uuid::parse_str(file_name).is_err() {
-                    log::warn!("Pattern mismatch {}", path.display());
-                    delete_file(&path);
-                    return None;
-                }
-                // In case we can't process the file we just ignore it.
-                if let Some(request) = process_ping_file(file_name, &path) {
-                    // Get the modified date of the file, which will later be used
-                    // for sorting the resulting vector.
-                    let modified_date = fs::metadata(&path).and_then(|data| data.modified());
-                    return Some((modified_date, request));
-                }
-            };
-            None
-        })
-        .collect();
-
-    // Sort by `modified_date`.
-    pending_pings.sort_by(|(a, _), (b, _)| {
-        // We might not be able to get the modified date for a given file,
-        // in which case we just put it at the end.
-        if let (Ok(a), Ok(b)) = (a, b) {
-            a.partial_cmp(b).unwrap()
+    /// Get the pending pings directory path.
+    pub fn get_dir(&self) -> PathBuf {
+        self.data_path.join("pending_pings")
+    }
+
+    /// Attempts to delete a ping file.
+    ///
+    /// ## Arguments
+    ///
+    /// * `uuid` - The UUID of the ping file to be processed
+    ///
+    /// ## Panics
+    ///
+    /// Won't panic if unable to delete the file.
+    pub fn delete_file(&self, uuid: &str) {
+        let path = self.get_dir().join(uuid);
+        match fs::remove_file(&path) {
+            Err(e) => log::error!("Error deleting file {}. {}", path.display(), e),
+            _ => log::info!("Files was deleted {}", path.display()),
+        };
+    }
+
+    /// Reads a ping file and returns a `PingRequest` from it.
+    ///
+    /// If the file is not properly formatted, it will be deleted.
+    ///
+    /// ## Arguments
+    ///
+    /// * `uuid` - The UUID of the ping file to be processed
+    ///
+    /// ## Panics
+    ///
+    /// Will panic if unable to read the file.
+    pub fn process_file(&self, uuid: &str) -> Option<PingRequest> {
+        let path = self.get_dir().join(uuid);
+        let file = File::open(path).expect("Should be able to read ping file.");
```

Comment:
And if not?
It might be missing/inaccessible/something else. Let's handle that gracefully and return None.

```suggestion
        let file = match File::open(path).expect("Should be able to read ping file.") {
        	Ok(file) => file,
        	Err(_) => return None,
        };
```